test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/CharacterTest.scala

```diff
@@ -383,6 +383,27 @@ class CharacterTest {
     assertFalse(Character.isDigit('\uFBFC'))
   }
 
+  @Test def codePointCount(): Unit = {
+    assertEquals(1, Character.codePointCount("\uD800\uDC00", 0, 2))
+    assertEquals(1, Character.codePointCount("\uD800\uDC01", 0, 2))
+    assertEquals(1, Character.codePointCount("\uD801\uDC01", 0, 2))
+    assertEquals(1, Character.codePointCount("\uDBFF\uDFFF", 0, 2))
+    assertEquals(3, Character.codePointCount("a\uD800\uDC00b", 0, 4))
+    assertEquals(4, Character.codePointCount("a\uD800\uDC00b\uD800", 0, 5))
```

Comment:
Those tests cover only cases where `beginIndex = 0` and `endIndex = length`.

But again, we already have more complete tests for that in `StringTest`:
https://github.com/scala-js/scala-js/blob/bb4835763b9316c8370bc00b6bd4f6c0653a0e95/test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/StringTest.scala#L176-L195
We should use the same tests.

We should also test with `CharSequence`s that are *not* strings. We have [`WrappedStringCharSequence`](https://github.com/scala-js/scala-js/blob/master/test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/WrappedStringCharSequence.scala) in the test suite for that purpose. So we should duplicate all the tests with instances of `WrappedStringCharSequence`.